routing/dht/dht.go

```diff
@@ -279,9 +289,24 @@ func (dht *IpfsDHT) findProvidersSingle(ctx context.Context, p peer.ID, key key.
 
 // nearestPeersToQuery returns the routing tables closest peers.
 func (dht *IpfsDHT) nearestPeersToQuery(pmes *pb.Message, count int) []peer.ID {
-	key := key.Key(pmes.GetKey())
-	closer := dht.routingTable.NearestPeers(kb.ConvertKey(key), count)
-	return closer
+	if pmes.Key != nil {
+		return dht.routingTable.NearestPeers(kb.ConvertKey(key.Key(pmes.GetKey())), count)
+	} else {
+		closer := make(map[peer.ID]struct{})
+		for _, sk := range pmes.GetKeys() {
+			np := dht.routingTable.NearestPeers(kb.ConvertKey(key.Key(sk)), count)
+			for _, p := range np {
+				closer[p] = struct{}{}
+			}
+		}
+
+		var out []peer.ID
+		for p, _ := range closer {
+			out = append(out, p)
+		}
+
+		return out
```

Comment:
I think you don't quite understand what this function is used for... This is used by a node who is asked for closer peers to select peers to reply with.